prometheus/append_test.go

```diff
@@ -0,0 +1,93 @@
+package prometheus
+
+import (
+	"testing"
+	"time"
+)
+
+var testMetrics = []struct {
+	scenario string
+	metric   metric
+	string   string
+}{
+	{
+		scenario: "simple counter metric",
+		metric: metric{
+			mtype: counter,
+			name:  "hello_world",
+			value: 42,
+		},
+		string: `# TYPE hello_world counter
+hello_world 42
+`,
+	},
+
+	{
+		scenario: "simple gauge metric",
+		metric: metric{
+			mtype: gauge,
+			name:  "hello_world",
+			value: 42,
+		},
+		string: `# TYPE hello_world gauge
+hello_world 42
+`,
+	},
+
+	{
+		scenario: "simple histogram metric",
+		metric: metric{
+			mtype:  histogram,
+			name:   "hello_world_bucket",
+			value:  42,
+			labels: labels{{"le", "0.5"}},
+		},
+		string: `# TYPE hello_world histogram
+hello_world_bucket{le="0.5"} 42
+`,
+	},
+
+	{
+		scenario: "counter metric with help, floating point value, labels, and timestamp",
+		metric: metric{
+			mtype:  counter,
+			scope:  "global",
+			name:   "hello_world",
+			help:   "This is a great metric!\n",
+			value:  0.5,
+			labels: []label{{"question", "\"???\"\n"}, {"answer", "42"}},
+			time:   time.Date(2017, 6, 4, 22, 12, 0, 0, time.UTC),
+		},
+		string: `# HELP global_hello_world This is a great metric!\n
```

Comment:
Yes, and the `# TYPE` and `# HELP` comments are interpreted by prometheus to get the type and help message of a metric.